ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,229 @@
+
+RULE: If we have released a package FOO-X.Y.Z, then the release-FOO-X.Y.x branch MUST exist and its first-parent history MUST include the commit released as FOO-X.Y.Z.
+For example, when releasing version 2.3.0 of a package named FOO, we must create the branch named release-FOO-2.3.x.
+And when we later release 2.3.1 of FOO, then we'd need to advance the release-FOO-2.3.x branch to point at the commit being released as FOO-2.3.1.
+See the "Release Branch Example" below for a more thorough example.
+
+There are two kinds of work on release branches.
+Most of the time, that work is immitating some work that was done on the main branch.
+EG we fixed a bug on the main branch and we're backporting that fix onto a release branch (hopefully it's a trivial cherry-pick).
+Rarely, though, there will be fresh work done on a release branch.
+EG Consider the following possible timeline.
+
+- We release FOO-2.4.1 on Monday, and FOO-2.4.1 completely reworked some functionality.
+- We then realize on Tuesday that there was a severe bug in the old functionality.
+- We fix the bug by merging a PR that targets the FOO-2.3.x release branch (since the bug no longer exists on the main branch!)
+- We release FOO-2.3.7 on Wednesday.
+
+*Remark*.
+Not every first-parent commit in the history of the release branch was announced as a release of a package.
+We will be merging multiple PRs into the release branch in order to prepare the next release from it, so some commits will just be the intermediates between two releases.
+
+*Remark*.
+The release branch for the greatest MAJOR.MINOR version of a package is somewhat degenerate.
+It's either equal to or a prefix of the main branch.
+The only time it MUST be updated to the tip of the main branch is when we cut a release of the package from the main branch, to satisfy the rule above about all commits that were released as some version X.Y.Z being on the release-X.Y.x branch.
+
+## Rules for PRs
+
+We classify each PR as either a _fresh_ PR or as a _backport_ PR.
+
+- A fresh PR is doing work that has never been done before (new feature, new bugfix, etc).
+  Except in the rare circumstances discussed in the previous section, every fresh PR will target the main branch.
+- The primary objective of a backport PR is merely to immitate some fresh PR.
+
+The rules are then as follows.
+
+- RULE: A fresh PR MUST target the main branch, if possible.
+- RULE: A PR MUST NOT be merged into a branch unless it should be included in the next release from that branch.
+- We maintain a changelog for each package (one for FOO and one for BAR).
+- RULE: A PR MUST add a pending changelog entry for each package that it alters.
+- RULE: Each pending changelog entry MUST at the very least classify the alteration to that package as a _patch_ level change (eg a bugfix), a _minor_ change (eg a non-breaking interface change), or a _major_ change (eg a breaking interface change).
```

Comment:
Do we want to add a RULE that merge bot such as [Mergify](https://docs.mergify.com/actions/backport/#backport-action) must be used if possible (i.e. no conflicts)?

Reply:
This document is only about which branches&versions should exist and how they should relate. How you actually merge the PRs seems beyond the scope of that. What do you think?